### PR TITLE
Blobstore Integration

### PR DESCRIPTION
**Pull request by lailagamaleldin**
This PR lets users upload a profile picture onto their profile using Blobstore. Right now only a user can see their own picture. The next step is to display the pictures of users' matches although this shouldn't be too challenging since all the infrastructure needed to support Blobstore has been set up in this PR.
image

### CODE REVIEW

File: step/src/main/java/com/google/sps/servlets/ImageUploadServlet.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    UserService userService = UserServiceFactory.getUserService();
+    String id = userService.getCurrentUser().getUserId();
+    String imageBlobKey = filterAndUploadImageBlobstore(request);
+
+    if (imageBlobKey == null) {
+      System.err.println("Error occured while uploading user's image");
+      // TODO: Display some message on the screen telling the user to check the file
+      //       they uploaded and try again.
+    } else {
+      System.out.println("not null");
```
Comment (adamsmlsn) [inline step/src/main/java/com/google/sps/servlets/ImageUploadServlet.java]: I think this line can be taken out now

File: step/src/main/webapp/profile.jsp [hunk 2]
```diff
@@ -88,12 +88,16 @@ limitations under the License.
     <div class="container">
         <div class="sub-container" id="list-selection">
             <h2> Your Profile </h2>
-            <div id="profile-pic"> <img src="avatar.png" alt="Profile Picture"> </div>
```
Comment (lilbroadam) [inline step/src/main/webapp/profile.jsp:91]: Is this image still displayed when the user hasn't uploaded a profile picture?

Comment (lailagamaleldin) [reply step/src/main/webapp/profile.jsp:91]: Yes, it's set in Javascript where if Datastore doesn't return a url for the user's image then it's set to avatar.png.

**Review by nganttran — APPROVED** [review]

**Review by adamsmlsn — APPROVED** [review]

**Comment by adamsmlsn** [comment]
FYI: The client id/secret was reset since I wasn't able to completely remove it from git history. The client id/secret found in the repo are no longer valid and API requests using them will be rejected.

**Pull request merged**